src/content/docs/zh-tw/basics/layouts.mdx

```diff
@@ -1,101 +1,269 @@
+  </body>
+</html>
+```
+
+### Markdown 版面參數
+
+Markdown/MDX 版面能透過 `Astro.props` 存取下列資訊：
+
+- **`file`**：檔案的絕對路徑 (例如 `/home/user/projects/.../file.md`)。
+- **`url`**：如果是頁面的話，即頁面網址（例如 `/zh-tw/guides/markdown-content`）。
+- **`frontmatter`**：Markdown 或 MDX 文件中的所有 frontmatter。
+  - **`frontmatter.file`**：同最上層的 `file` 屬性。
+  - **`frontmatter.url`**：同最上層的 `url` 屬性。
+- **`headings`**：Markdown 或 MDX 文件中的標題（`h1 -> h6`）列表，包含對應的 metadata。其型別為 `{ depth: number; slug: string; text: string }[]`。
+- **（Markdown 專屬）`rawContent()`**：取得 Markdown 原始內容的函式，回傳值格式為字串。
+- **（Markdown 專屬）`compiledContent()`**：取得 Markdown 編譯後內容的函式，回傳值格式為 HTML 字串。
+
+舉例來說，Markdown 撰寫的部落格文章可以把下列 `Astro.props` 物件傳給它的版面：
+
+```js
+Astro.props = {
+  file: "/home/user/projects/.../file.md",
+  url: "/zh-tw/guides/markdown-content/",
+  frontmatter: {
+    /** 部落格文章的 frontmatter */
+    title: "Astro 0.18 Release",
+    date: "Tuesday, July 27 2021",
+    author: "Matthew Phillips",
+    description: "Astro 0.18 is our biggest release since Astro launch.",
+    /** 產生的值 */
+    file: "/home/user/projects/.../file.md",
+    url: "/zh-tw/guides/markdown-content/"
+  },
+  headings: [
+    {
+      "depth": 1,
+      "text": "Astro 0.18 Release",
+      "slug": "astro-018-release"
+    },
+    {
+      "depth": 2,
+      "text": "Responsive partial hydration",
+      "slug": "responsive-partial-hydration"
+    }
+    /* ... */
+  ],
+
+  /** 只能在 Markdown 使用  */
+  rawContent: () => "# Astro 0.18 Release\nA little over a month ago, the first public beta [...]",
+  compiledContent: () => "<h1>Astro 0.18 Release</h1>\n<p>A little over a month ago, the first public beta [...]</p>",
+}
+```
+
+:::note
+Markdown/MDX 版面也能透過 `Astro.props` 存取檔案的[匯出屬性](/zh-tw/guides/markdown-content/#exported-properties)，不過其中**有些差異**：
+
+* 標題資訊（即 `h1 -> h6` 元素）透過 `headings` 陣列存取，而非 `getHeadings()` 函式。
+* `file` 和 `url` *也*可在巢狀 `frontmatter` 屬性下存取（即 `frontmatter.url` 和 `frontmatter.file`）。
+* 無法存取在 frontmatter 以外宣告的數值（例如 MDX 的 `export` 表達式）。如有類似需求，請考慮[匯入版面](#手動匯入版面mdx)。
+:::
+
+### 手動匯入版面（MDX）
+
+需要傳遞資訊到 MDX 版面，但該版面不存在（或無法存在）frontmatter 時，可以匯入 [`<Layout />` 元件](/zh-tw/basics/layouts/)，再像其他元件一樣透過參數傳遞給它：
+
+```mdx title="src/pages/posts/first-post.mdx" ins={6} del={2} /</?BaseLayout>/ /</?BaseLayout title={frontmatter.title} fancyJsHelper={fancyJsHelper}>/
+---
+layout: ../../layouts/BaseLayout.astro
+title: 'My first MDX post'
+publishDate: '21 September 2022'
+---
+import BaseLayout from '../../layouts/BaseLayout.astro';
+
+function fancyJsHelper() {
+  return "Try doing that with YAML!";
+}
+
+<BaseLayout title={frontmatter.title} fancyJsHelper={fancyJsHelper}>
+  Welcome to my new Astro blog, using MDX!
+</BaseLayout>
+```
+
+如此一來，版面便能透過 `Astro.props` 存取數值，而 MDX 內容則會嵌入到包含 `<slot />` 的頁面中：
+
+```astro /{?title}?/ "fancyJsHelper" "{fancyJsHelper()}"
+---
+// src/layouts/BaseLayout.astro
+const { title, fancyJsHelper } = Astro.props;
+---
+<!-- -->
+<h1>{title}</h1>
+<slot /> <!-- your content is injected here -->
```

Comment:
This comment can be translated.

Reply:
Thanks for spotting this! Comment translated: fe96551219952e3c15397b90d818c40684369dcb